### PR TITLE
MKDocs integration

### PR DESCRIPTION
**Pull request by omri374**
# MKDocs integration

## What are you trying to address

Creating a static website for the playbook based on [mkdocs](https://www.mkdocs.org/) and [mkdocs-material](https://squidfunk.github.io/mkdocs-material/). 
Demo can be found here: https://omri374.github.io/code-with-engineering-playbook/

### To test locally
1. Create a virtual Python env
2. Install the dependencies (`pip install -r requirements-docs.txt`)
3. Run `mkdocs serve`

## Description of new changes

- Moved content into the `docs` folder.
- Minor changes to content if necessary for proper rendering on mkdocs.
- Removed relative links to docs in the repo but outside the docs folder (e.g. in the .github folder).
- Added `mkdocs.yml` which specifies the mkdocs behavior.
- Added a `requirements-docs.txt` file with the required dependencies for generating the docs.

## Still missing
- [ ] Align active PRs accordingly (mainly changing the location of files from the root of the repo to the docs folder).
- [X] Reviewing the content to make sure everything is rendered correctly on the mkdocs website.
- [X] Adding Github Action for creating and uploading the website to https://microsoft.github.io/code-with-engineering-playbook.
- [X] Freeze versions on `requirements-docs.txt`
- [X] Fix linting issues 

Relevant issue: #549

## For all pull requests

- [X] Link to the issue you are solving (so it gets closed)
- [X] Label the pull request with the appropriate area(s)
- [X] Assign potential reviewers (you may also want to contact them on Teams to ensure timely reviews)
- [X] Assign the project - Engineering Playbook Backlog
- [X] Assign the pull request to yourself

### CODE REVIEW

File: docs/index.md
Comment (fnocera) [inline docs/index.md:3]: I think this is causing a `markdownlint` error because title is not a level 1 heading. You could add `<!-- markdownlint-disable-file MD041 -->` to the top of the page? 
Also might need to add a line before the `title:Home` line for markdownlint too.

File: docs/index.md
Comment (fnocera) [inline docs/index.md:17]: Markdownlint is complaining about the titles, we might have to change the markdownlint.yaml to include: `"MD003": { "style": "atx_closed" }`

File: mkdocs.yml
Comment (fnocera) [inline mkdocs.yml:29]: It seems to deploy the site we just have to use the `mkdocs gh-deploy` command as documented here https://www.mkdocs.org/user-guide/deploying-your-docs/#project-pages

File: docs/index.md
Comment (omri374) [reply docs/index.md:3]: Sure, I saw it failed the lint but had to add this line to change the title on the sidebar.

File: mkdocs.yml
Comment (omri374) [reply mkdocs.yml:29]: Yes, plus the installation of dependencies:
```sh
pip install --upgrade pip      
pip install wheel                                 
pip install -r requirements-docs.txt
mkdocs gh-deploy
```

File: docs/index.md
Comment (omri374) [reply docs/index.md]: Removed this line to avoid linting issues. It's not mandatory.

File: docs/index.md
Comment (omri374) [reply docs/index.md:12]: Fixed (removing the yaml line fixed it)

**Comment by TessFerrandez** [comment]
Looks great!!!

**Review by TessFerrandez — APPROVED** [review]

**Comment by fnocera** [comment]
Looking at the link checking errors these are the ones failing:

FILE: ./CONTRIBUTING.md
[✖] code-reviews/recipes/Markdown.md
[✖] ./code-reviews/pull_request_template.md
[✖] ./source-control/contributing/merge-strategies.md

16 links checked.

ERROR: 3 dead links found!
[✖] code-reviews/recipes/Markdown.md → Status: 400
[✖] ./code-reviews/pull_request_template.md → Status: 400
[✖] ./source-control/contributing/merge-strategies.md → Status: 400

FILE: ./README.md
[✖] ./CSE.md
[✖] ENG-FUNDAMENTALS-CHECKLIST.md
[✖] SPRINT-STRUCTURE.md

21 links checked.

ERROR: 3 dead links found!
[✖] ./CSE.md → Status: 400
[✖] ENG-FUNDAMENTALS-CHECKLIST.md → Status: 400
[✖] SPRINT-STRUCTURE.md → Status: 400

**Comment by fnocera** [comment]
We can say that 
- Aligning PRs accordingly can come after the first version of this PR is merged in. 
- Consider creating a new domain name for the playbook.

**Comment by omri374** [comment]
@fnocera  I'm working on the linting issues.

**Comment by omri374** [comment]
@fnocera @TessFerrandez all lint errors are fixed. PR is ready for review and merge.

**Comment by fnocera** [comment]
@brockneedscoffee flagging for your review as well

**Review by fnocera — CHANGES_REQUESTED** [review]
Approved - pending review that the new documentation section is merged in/added to the PR.  

Thanks so much for this huge contribution @omri374, this will make the playbook look and feel so much more user friendly.

**Review by fnocera — APPROVED** [review]
Approved - pending review that the new documentation section is merged in/added to the PR.

Thanks so much for this huge contribution @omri374, this will make the playbook look and feel so much more user friendly.

**Pull request merged**
